iodata/formats/gromacs.py

```diff
@@ -0,0 +1,145 @@
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 3
+# of the License, or (at your option) any later version.
+#
+# IODATA is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>
+# --
+"""gro file format.
+
+Files with the gro file extension contain a molecular structure in Gromos87 format.
+gro files can be used as trajectory by simply concatenating files.
+
+http://manual.gromacs.org/current/reference-manual/file-formats.html#gro
+
+"""
+
+
+from typing import Tuple, Iterator
+
+import numpy as np
+
+from ..docstrings import (document_load_one, document_load_many)
+from ..utils import nanometer, picosecond, LineIterator
+
+
+__all__ = []
+
+
+PATTERNS = ['*.gro']
+
+
+@document_load_one('GRO', ['atcoords', 'atffparams', 'cellvecs', 'extra', 'title'])
+def load_one(lit: LineIterator) -> dict:
+    """Do not edit this docstring. It will be overwritten."""
+    while True:
+        try:
+            data = _helper_read_frame(lit)
+        except StopIteration:
+            break
+        if data is not None:
+            title = data[0]
+            time = data[1]
+            resnums = np.array(data[2])
+            resnames = np.array(data[3])
+            attypes = np.array(data[4])
+            atcoords = data[5]
+            velocities = data[6]
+            cellvecs = data[7]
+            atffparams = {
+                'attypes': attypes,
+                'resnames': resnames,
+                'resnums': resnums
+            }
+            extra = {
+                'time': time,
+                'velocities': velocities
+            }
+            result = {
+                'atcoords': atcoords,
+                'atffparams': atffparams,
+                'cellvecs': cellvecs,
+                'extra': extra,
+                'title': title,
+            }
+            return result
+    raise lit.error('Bad format. Gromacs gro file could not be read.')
+
+
+@document_load_many('GRO', ['atcoords', 'atffparams', 'cellvecs', 'extra', 'title'])
+def load_many(lit: LineIterator) -> Iterator[dict]:
+    """Do not edit this docstring. It will be overwritten."""
+    # gro files can be used as trajectory by simply concatenating files,
+    # making it trivial to load many frames.
+    while True:
+        try:
+            yield load_one(lit)
+        except IOError:
+            return
+
+
+def _helper_read_frame(lit: LineIterator) -> Tuple:
+    """Read one frame."""
+    # Read the first line, get the title and try to get the time.
+    # Time field is optional.
+    line = next(lit)
+    title = line.split(',')[0] if 't=' in line else line[:-1]
```

Comment:
If time is not given in the title, I am confused why line is reversed...
```suggestion
    title = line.split(',')[0] if 't=' in line else line
```

Reply:
It takes the complete line except the new line character. Just aesthetics. I can change it 👍

Reply:
Sorry, for some reason I thought you are using `line[::-1]` which reverses the string, but you had only one `:`. Removing the new line character is good, and you shouldn't remove it.